### DISCUSSION TITLE
Ideas for autocompletion

### DISCUSSION THREAD

**Discussion by geanruca** [fish-shell/fish-shell#9773] [Ideas]
Hello! I have been working with fish for a few months and love it. I would like to contribute to building a few improvements (If we already have them, I'm sorry 😢)

1. For Git auto-completion: Integration of this suggestion after writing git push 
` --set-upstream origin <new-branch> `
 and do it automatically when we are in a new branch. Actually, I receive a suggestion of set-upstream when I'm in a new branch, but the suggestion is wrong because it always wants to set the origin in my last working branch, and not in the current and new one.

2. Suggestions: if I have a `git checkout -b` in my history, recommend the full command in the history without the -b (which is usually used only once in general)

I would love to work on this project, please let me know if this is already integrated!

**Comment by faho** [comment]
The autosuggestions - that greyed-out part of the commandline - are currently not command-specific and (mostly) unscriptable. This allows them to easily and quickly run in the background. I am not enthusiastic about adding special-cases for commands here.

The completions - that thing you get when pressing TAB - are scriptable, but they only ever offer one token, and they try to offer all possibilties.

**Reply by geanruca** [reply]
Thanks for answering 😁 it's pretty much clear to me
